src/main/kotlin/org/rust/ide/inspections/RsLintInspection.kt

```diff
@@ -0,0 +1,106 @@
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections
+
+import com.intellij.codeInspection.ContainerBasedSuppressQuickFix
+import com.intellij.codeInspection.ProblemDescriptor
+import com.intellij.codeInspection.SuppressQuickFix
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiNamedElement
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.*
+
+abstract class RsLintInspection : RsLocalInspectionTool() {
+
+    protected abstract val lint: RsLint
+
+    override fun isSuppressedFor(element: PsiElement): Boolean {
+        if (super.isSuppressedFor(element)) return true
+        return lint.levelFor(element) == RsLintLevel.ALLOW
+    }
+
+    // TODO: fix quick fix order in UI
+    override fun getBatchSuppressActions(element: PsiElement?): Array<SuppressQuickFix> {
+        val fixes = super.getBatchSuppressActions(element).toMutableList()
+        if (element == null) return fixes.toTypedArray()
+        element.ancestors.forEach {
+            val action = when (it) {
+                is RsLetDecl,
+                is RsFieldDecl,
+                is RsEnumVariant,
+                is RsItemElement,
+                is RsFile -> {
+                    var target = when (it) {
+                        is RsLetDecl -> "statement"
+                        is RsFieldDecl -> "field"
+                        is RsEnumVariant -> "enum variant"
+                        is RsStructItem -> "struct"
+                        is RsEnumItem -> "enum"
+                        is RsFunction -> "fn"
+                        is RsTypeAlias -> "type"
+                        is RsConstant -> "const"
+                        is RsModItem -> "mod"
+                        is RsImplItem -> "impl"
+                        is RsTraitItem -> "trait"
+                        is RsUseItem -> "use"
+                        is RsFile -> "file"
+                        else -> null
+                    }
+                    if (target != null) {
+                        val name = (it as? PsiNamedElement)?.name
+                        if (name != null) {
+                            target += " $name"
+                        }
+                        RsSuppressQuickFix(it as RsDocAndAttributeOwner, lint, target)
+                    } else {
+                        null
+                    }
+                }
+                is RsExprStmt-> {
+                    val expr = it.expr
+                    if (expr is RsOuterAttributeOwner) RsSuppressQuickFix(expr, lint, "statement") else null
+                }
+                else -> null
+            }
+            if (action != null) {
+                fixes += action
+            }
+        }
+        return fixes.toTypedArray()
+    }
+}
+
+private class RsSuppressQuickFix(
+    private val suppressAt: RsDocAndAttributeOwner,
```

Comment:
Possible memory leak here (quick fixes shouldn't store references to PSI)

Reply:
Oh, thanks!
Will fix it asap
Good thing: now I know that DevKit plugin has inspection to detect such mistakes and it's great!